### PR TITLE
Add CIS 3.1.1

### PR DESCRIPTION
**Pull request by satya-rajesh**
Added and verified

### CODE REVIEW

File: roles/cis/tasks/3/3.1.1.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (keerthanakalyan) [inline roles/cis/tasks/3/3.1.1.yml:1]: Does this call sysctl command add the entry to sysctl.conf file if value is not the expected value and do a flush? Please refer this https://gitlab.eng.vmware.com/core-build/photon-hardening/commit/cdb23e5ab12bee7fba5213894ba53583400073d0 incase you have not seen the previous script implementaion on how we handle all sysctl based configurations and we generally write the photon specific sysctl configs to /etc/sysctl.d/50-security-hardening.conf not to /etc/sysctl.conf just to differentiate what is been set by photon hardening.

File: roles/cis/tasks/3/verify/3.1.1.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+- name: "3.1.1 Verify if IP forwarding is disabled (Scored)"
+  block:
+    - name: "3.1.1 verify if net.ipv4.ip_forward is set to 0"
+      command: grep -Fxq "net.ipv4.ip_forward=0" /etc/sysctl.conf
```
Comment (keerthanakalyan) [inline roles/cis/tasks/3/verify/3.1.1.yml]: This check wont be sufficient for verifying whether ip_forward is disabled. Even though this value is in /etc/sysctl.conf there might be cases where this value is not properly applied. Better way is to  check the runtime proc values or use sysctl command to query for this and check the return value.

File: roles/cis/tasks/3/verify/3.1.1.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      register: output_1
+    - fail:
+        msg: "3.1.1 failed to verify if IP forwarding is disabled, net.ipv4.ip_forward is not set to 0"
+      when: output_1.rc != 0
+    - name: "3.1.1 verify if net.ipv6.conf.all.forwarding is set to 0"
+      command: grep -Fxq "net.ipv6.conf.all.forwarding=0" /etc/sysctl.conf
```
Comment (keerthanakalyan) [inline roles/cis/tasks/3/verify/3.1.1.yml]: This check wont be sufficient for verifying whether ip_forward is disabled. Even though this value is in /etc/sysctl.conf there might be cases where this value is not properly applied. Better way is to check the runtime proc values or use sysctl command to query for this and check the return value.

File: roles/cis/tasks/3/3.1.1.yml (hunk 1 above)
Comment (satya-rajesh) [reply roles/cis/tasks/3/3.1.1.yml:1]: Thanks for the review.
unfortunately in chroot, sysctl commands are throwing errors:
```shell
    if [ "$1" == "both" ]; then
       sysctl -w $IPROUTEFLUSH > /dev/null
       sysctl -w $IPV6ROUTEFLUSH > /dev/null
    elif [ "$1" == "ipv4" ]; then
       sysctl -w $IPROUTEFLUSH > /dev/null
    elif [ "$1" == "ipv6" ]; then
       sysctl -w $IPV6ROUTEFLUSH > /dev/null
    fi
```
I will add another block for executing shell commands in localhost.
In localhost the sysctl shell commands are working.

recommendation was to use sysctl module which by default is writing to /etc/sysctl.conf:
https://docs.ansible.com/ansible/latest/modules/sysctl_module.html
I will investigate to write to /etc/sysctl.d/50-security-hardening.conf

File: roles/cis/tasks/3/verify/3.1.1.yml (hunk 2 above)
Comment (satya-rajesh) [reply roles/cis/tasks/3/verify/3.1.1.yml]: I will add another block for sysctl shell commands verification with localhost tag

File: roles/cis/tasks/3/3.1.1.yml (hunk 1 above)
Comment (ppadmavilasom) [reply roles/cis/tasks/3/3.1.1.yml:1]: you can change the file destination of sysctl module. 
as @keerthanakalyan points out, the verification of this rule will only verify file contents if done in chroot. this is okay. please add comment in verification stating the same.
verification for production should be done via inspec or similar.

Comment (satya-rajesh) [reply roles/cis/tasks/3/3.1.1.yml:1]: Done, the changes are made to apply the configuration to /etc/sysctl.d/50-security-hardening.conf and verify the same

**Review by ppadmavilasom — COMMENTED** [review]
@satya-rajesh ci build is only running 1.7.1.6. can you check why that is? expected ci to run all. can you check with dev branch if it is running all existing rules and verification and compare with your PR?

**Comment by satya-rajesh** [comment]
> @satya-rajesh ci build is only running 1.7.1.6. can you check why that is? expected ci to run all. can you check with dev branch if it is running all existing rules and verification and compare with your PR?

sysctl is not found as CIS 3.1.1 requires it, shall make changes to install procps-ng in chroot along with this change

**Pull request merged**